packages/store/src/StoreCore.sol

```diff
@@ -0,0 +1,524 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+import { Utils } from "./Utils.sol";
+import { Bytes } from "./Bytes.sol";
+import { SchemaType, getStaticByteLength, getElementByteLength } from "./Types.sol";
+import { Storage } from "./Storage.sol";
+import { Memory } from "./Memory.sol";
+import { console } from "forge-std/console.sol";
+import { Schema } from "./Schema.sol";
+import { PackedCounter } from "./PackedCounter.sol";
+import { Buffer, Buffer_ } from "./Buffer.sol";
+import { HooksTable, tableId as HooksTableId } from "./tables/HooksTable.sol";
+import { IStoreHook } from "./IStore.sol";
+
+// TODO
+// - Turn all storage pointer to uint256 for consistency (uint256 is better than bytes32 because it's easier to do arithmetic on)
+// - Change Storage library functions to make it clearer which argument is offset and which is length
+// - Streamline naming in Storage and Memory libraries (probably just use load and store instead of read and write?)
+// - Move all internal functions to a separate StoreCoreInternal library to discourage using it
+
+library StoreCore {
+  // note: the preimage of the tuple of keys used to index is part of the event, so it can be used by indexers
+  event MudStoreSetRecord(bytes32 table, bytes32[] key, bytes data);
+  event MudStoreSetField(bytes32 table, bytes32[] key, uint8 schemaIndex, bytes data);
+  event MudStoreDeleteRecord(bytes32 table, bytes32[] key);
+
+  bytes32 internal constant SLOT = keccak256("mud.store");
+  bytes32 internal constant SCHEMA_TABLE = keccak256("mud.store.table.schema");
+
+  error StoreCore_TableAlreadyExists(bytes32 table);
+  error StoreCore_TableNotFound(bytes32 table);
+  error StoreCore_NotImplemented();
+  error StoreCore_InvalidDataLength(uint256 expected, uint256 received);
+  error StoreCore_NoDynamicField();
```

Comment:
I noticed some of our error names use `_` and recently others do not. We should prob converge on some pattern.

I also wonder about similarly prefixing errors with `Mud`. Seems like that's the direction folks are going with ERC721, etc. errors because it helps with overlapping names.

That said, I'm now seeing `event MudStore...(...)` in the context of `StoreCore` and wondering again about no prefix vs. prefixing everything internal. 🙈  Can't decide!